# 异常

```py
try:
    执行代码
expect:
    出现错误的处理
```

一个例子
```py
try:
    num=int(input("input an integer number:"))
except:
    print("input error!")
```

## 根据错误类型捕获异常

```py
try:
    执行语句
except 错误类型1:
    执行语句
except (错误类型1,错误类型2):
    执行语句
except Exception as result:
    print("未知错误%s" % result)
```

一个例子

```py
# 输入一个数作为除数与 12 进行运算

try:
    num=int(input("input an integer number:"))
    result=12/num
    print(result)

except ZeroDivisionError:
    print("0 不能作为除数!")
    
except ValueError:
    print("请输入一个整数")

except Exception as result:
    print("未知错误%s" % result)
```